### PR TITLE
secret-wrapper: handle writable home

### PR DESCRIPTION
**Pull request by stevekuznetsov**
There are a number of changes we make to the environment of the process
under test. Some occur in the `secret-wrapper` tool, while some form an
ugly Bash preable. As the content of these changes continues to grow and
become more complex, we should aim to make all of them in the Go tool.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @bbguimaraes 
/cc @alvaroaleman 

This is backwards  compatible and will work with *both* `secret-wrapper` and the shell. I'll remove the shell later.

### CODE REVIEW

File: cmd/secret-wrapper/main.go [hunk 1]
```diff
@@ -172,6 +172,18 @@ func execCmd(argv []string) error {
 	proc := exec.Command(argv[0], argv[1:]...)
 	proc.Stdout = os.Stdout
 	proc.Stderr = os.Stderr
+	// provide a writeable home so kubectl discovery can be cached
```
Comment (alvaroaleman) [inline cmd/secret-wrapper/main.go:175]: Can we rename the tool if it now does stuff that is entirely unrelated to secret wrapping?

Comment (stevekuznetsov) [reply cmd/secret-wrapper/main.go:175]: Sure, can I do that after I land the useful bits? :)

Comment (alvaroaleman) [reply cmd/secret-wrapper/main.go:175]: I am afraid that won't happen then :P Test failure looks genuine btw

**Comment by stevekuznetsov** [comment]
Needed for DPTP-1668

File: cmd/secret-wrapper/main.go [hunk 2]
```diff
@@ -172,6 +172,18 @@ func execCmd(argv []string) error {
...
+	home, set := os.LookupEnv("HOME")
+	needHome := !set
+	if set {
+		if err := syscall.Access(home, syscall.O_RDWR); err != nil {
+			// $HOME is set but not writeable
+			needHome = true
+		}
+	}
+	if needHome {
+		proc.Env = append(os.Environ(), "HOME=/alabama")
```
Comment (bbguimaraes) [inline cmd/secret-wrapper/main.go:185]: To save time for other potential paranoid reviewers:
```
// If Env contains duplicate environment keys, only the last
// value in the slice for each duplicate key is used.
```

File: cmd/secret-wrapper/main.go [hunk 3]
```diff
@@ -172,6 +172,18 @@ func execCmd(argv []string) error {
...
+	}
```
Comment (bbguimaraes) [inline cmd/secret-wrapper/main.go:186]: This function will get quite involved, specially with the following PR.  Can we leave only the execution bits here and move the new code to a separate one?  Adding the environment as a parameter looks like a good interface.

File: cmd/secret-wrapper/main.go (hunk 2 above)
Comment (stevekuznetsov) [reply cmd/secret-wrapper/main.go:185]: Fixed in the following PR

File: cmd/secret-wrapper/main.go (hunk 3 above)
Comment (stevekuznetsov) [reply cmd/secret-wrapper/main.go:186]: Sure

**Comment by stevekuznetsov** [comment]
```
         * 3x kubelet: Failed to create pod sandbox: rpc error: code = DeadlineExceeded desc = context deadline exceeded 
```

/test e2e
